component/src/main/java/org/exoplatform/rhmanagement/services/UserDataService.java

```diff
@@ -0,0 +1,316 @@
+/*
+ *
+ *  * Copyright (C) 2003-2016 eXo Platform SAS.
+ *  *
+ *  * This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Affero General Public License
+ *  as published by the Free Software Foundation; either version 3
+ *  of the License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, see<http://www.gnu.org/licenses/>.
+ *
+ */
+package org.exoplatform.rhmanagement.services;
+
+import org.exoplatform.rhmanagement.dao.UserRHDataDAO;
+import org.exoplatform.rhmanagement.dto.EmployeesDTO;
+import org.exoplatform.rhmanagement.dto.UserRHDataDTO;
+import org.exoplatform.rhmanagement.entity.UserRHDataEntity;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Created by The eXo Platform SAS
+ */
+public class UserDataService {
+  private  final Logger LOG = LoggerFactory.getLogger(UserDataService.class);
+
+  private UserRHDataDAO userRHDataDAO;
+
+  private IdentityManager identityManager;
+  public UserDataService(IdentityManager identityManager) {
+    this.userRHDataDAO = new UserRHDataDAO();
+    this.identityManager = identityManager;
+  }
+
+  public UserRHDataDTO save(UserRHDataDTO entity) {
+    if (entity == null) {
+      throw new IllegalStateException("Parameter 'entity' is null");
+    }
+
+    UserRHDataEntity userRHDataEntity = null;
+      try {
+        UserRHDataDTO userRHDataDTO =getUserRHDataByUserId(entity.getUserId());
+        if(userRHDataDTO!=null){
+          userRHDataEntity = userRHDataDAO.update(convert(entity));
+        }else{
+          entity.setCreationDate(new Date());
+          userRHDataEntity = userRHDataDAO.create(convert(entity));
+        }
+      } catch (Exception e) {
```

Comment:
in the catch bloc, the same action is executed. The exception should be raised and another action should be executed instead of the previous one